thrust/detail/reference.inl

```diff
@@ -109,7 +110,7 @@ template<typename Element, typename Pointer, typename Derived>
   // XXX use null a reference for dispatching
   // XXX this assumes that the eventual invocation of
   // XXX get_value will not access system state
-  System *system = 0;
+  System system;
 
   return convert_to_value_type(system);
```

Comment:
I think this function should just call `get_value()` directly, right? I think the function `convert_to_value_type()` is no longer needed and should be eliminated.